bee-node/bee-plugin/src/message.rs

```diff
@@ -0,0 +1,793 @@
+//! not created by converting [`bee_message::Message`].
+
+pub use crate::grpc::Message;
+
+use crate::grpc::{
+    address::Kind as AddressKind, input::Kind as InputKind, output::Kind as OutputKind, payload::Kind as PayloadKind,
+    signature::Kind as SignatureKind, unlock_block::Kind as UnlockBlockKind, Address, ApplicationMessagePayload,
+    AssetBalance, AssetId, BeaconPayload, BlsAddress, BlsSignature, CollectiveBeaconPayload, Conflict, DataPayload,
+    DkgPayload, Ed25519Address, Ed25519Signature, EncryptedDeal, FpcPayload, IndexationPayload, Input, MessageId,
+    Output, OutputId, ParentsBlock, ParentsKind, Payload, ReferenceUnlock, Salt, SaltDeclarationPayload, Signature,
+    SignatureLockedAssetOutput, SignatureLockedSingleOutput, SignatureUnlock, Timestamp, TransactionEssence,
+    TransactionId, TransactionPayload, UnlockBlock, UtxoInput,
+};
+
+use std::convert::TryInto;
+
+impl From<&bee_message::Message> for Message {
+    fn from(message: &bee_message::Message) -> Self {
+        Self {
+            parents_blocks: message.parents_blocks().map(ParentsBlock::from).collect(),
+            issuer_public_key: message.issuer_public_key().to_vec(),
+            issue_timestamp: message.issue_timestamp(),
+            sequence_number: message.sequence_number(),
+            payload: message.payload().as_ref().map(Payload::from),
+            nonce: message.nonce(),
+            signature: message.signature().to_vec(),
+        }
+    }
+}
+
+#[allow(clippy::from_over_into)]
+impl Into<bee_message::Message> for Message {
+    fn into(self) -> bee_message::Message {
+        let mut builder = bee_message::MessageBuilder::default()
+            .with_issuer_public_key(self.issuer_public_key.try_into().unwrap())
+            .with_issue_timestamp(self.issue_timestamp)
+            .with_sequence_number(self.sequence_number)
+            .with_nonce(self.nonce)
+            .with_signature(self.signature.try_into().unwrap());
+
+        if let Some(payload) = self.payload {
+            builder = builder.with_payload(payload.into());
+        }
+
+        for parents_block in self.parents_blocks {
+            builder = builder.add_parents_block(parents_block.into());
+        }
+
+        builder.finish().unwrap()
+    }
+}
+
+impl From<&bee_message::parents::ParentsBlock> for ParentsBlock {
+    fn from(block: &bee_message::parents::ParentsBlock) -> Self {
+        Self {
+            kind: ParentsKind::from(block.parents_kind()).into(),
+            references: block.iter().map(MessageId::from).collect(),
+        }
+    }
+}
+
+#[allow(clippy::from_over_into)]
+impl Into<bee_message::parents::ParentsBlock> for ParentsBlock {
+    fn into(self) -> bee_message::parents::ParentsBlock {
+        // This and every other `unwrap` in an `Into::into` implementation is ok because we assume
+        // that `self` was created from a valid `bee_message::Message` and it has no missing
+        // fields.
```

Comment:
Is this necessary since you already have a note about it at module level ? This isn't even the first unwrap of the file so this is weird